### PR TITLE
Update for new message generation and iwyu

### PR DESCRIPTION
**Pull request by mjcarroll**
This moves the `private_msgs` proto files into their own directory to be used with https://github.com/gazebosim/gz-msgs/pull/368

It also updates many files to actually include the messages they use.

### CODE REVIEW

File: src/Server_TEST.cc [hunk 1]
```diff
@@ -25,6 +25,9 @@
 #include <gz/utils/ExtraTestMacros.hh>
 #include <sdf/Mesh.hh>
 
+#include <gz/msgs/server_control.pb.h>
+#include <gz/msgs/stringmsg_v.pb.h>
```
Comment (ahcorde) [inline src/Server_TEST.cc:29]: boolean and stringmsg

File: src/SimulationRunner_TEST.cc [hunk 2]
```diff
@@ -34,6 +34,9 @@
 
 
 #include "test_config.hh"
+
+#include "gz/msgs/clock.pb.h"
```
Comment (ahcorde) [inline src/SimulationRunner_TEST.cc:38]: GUI, SdfGeneratorConfig and StringMsg

File: src/cmd/ModelCommandAPI.cc [hunk 3]
```diff
@@ -32,6 +32,7 @@
 
 #include <gz/msgs/serialized.pb.h>
 #include <gz/msgs/stringmsg.pb.h>
+#include <gz/msgs/stringmsg_v.pb.h>
```
Comment (ahcorde) [inline src/cmd/ModelCommandAPI.cc:35]: add `SerializedStepMap` and remove `stringmsg`

File: src/gui/plugins/component_inspector/ComponentInspector.cc [hunk 4]
```diff
@@ -27,6 +27,9 @@
 #include <gz/math/Color.hh>
 #include <gz/math/SphericalCoordinates.hh>
 #include <gz/math/Vector3.hh>
+
+#include <gz/msgs/entity_plugin_v.pb.h>
```
Comment (ahcorde) [inline src/gui/plugins/component_inspector/ComponentInspector.cc:31]: Light, Boolean, Physics, Visual, SphericalCoordinates, Empty, EmptyPlugin_V

File: src/gui/plugins/playback_scrubber/PlaybackScrubber.cc [hunk 5]
```diff
@@ -18,6 +18,7 @@
 #include "PlaybackScrubber.hh"
 
 #include <gz/msgs/boolean.pb.h>
+#include <gz/msgs/log_playback_control.pb.h>
```
Comment (ahcorde) [inline src/gui/plugins/playback_scrubber/PlaybackScrubber.cc:21]: remote stringmsg

File: src/gui/plugins/visualize_contacts/VisualizeContacts.cc [hunk 6]
```diff
@@ -17,6 +17,7 @@
 
 #include <gz/msgs/contact.pb.h>
 #include <gz/msgs/contacts.pb.h>
+#include <gz/msgs/marker.pb.h>
```
Comment (ahcorde) [inline src/gui/plugins/visualize_contacts/VisualizeContacts.cc:20]: add Entity, boolean

File: src/network/NetworkManagerPrimary.cc [hunk 7]
```diff
@@ -27,8 +27,8 @@
 #include <gz/common/Util.hh>
 #include <gz/common/Profiler.hh>
 
-#include "msgs/peer_control.pb.h"
-#include "msgs/simulation_step.pb.h"
+#include "gz/sim/private_msgs/peer_control.pb.h"
+#include "gz/sim/private_msgs/simulation_step.pb.h"
```
Comment (ahcorde) [inline src/network/NetworkManagerPrimary.cc:31]: WorldStatistics, SerializedStateMap, SimulationStep

File: src/network/NetworkManagerPrimary.hh [hunk 8]
```diff
@@ -31,7 +31,7 @@
 #include <gz/sim/Entity.hh>
 #include <gz/transport/Node.hh>
 
-#include "msgs/simulation_step.pb.h"
+#include "gz/sim/private_msgs/simulation_step.pb.h"
```
Comment (ahcorde) [inline src/network/NetworkManagerPrimary.hh:34]: SerializedStateMap, SimulationStep, private_msgs::PerformerAffinity

File: src/network/NetworkManagerSecondary.cc [hunk 9]
```diff
@@ -22,7 +22,7 @@
 #include <gz/common/Util.hh>
 #include <gz/common/Profiler.hh>
 
-#include "msgs/peer_control.pb.h"
+#include "gz/sim/private_msgs/peer_control.pb.h"
```
Comment (ahcorde) [inline src/network/NetworkManagerSecondary.cc:25]: SerializedStateMap, private_msgs::SimulationStep

File: src/systems/camera_video_recorder/CameraVideoRecorder.cc [hunk 10]
```diff
@@ -20,6 +20,9 @@
 #include <string>
 #include <unordered_map>
 
+#include <gz/msgs/boolean.pb.h>
+#include <gz/msgs/video_record.pb.h>
```
Comment (ahcorde) [inline src/systems/camera_video_recorder/CameraVideoRecorder.cc:24]: image, time

File: src/systems/diff_drive/DiffDrive.cc [hunk 11]
```diff
@@ -17,7 +17,11 @@
 
 #include "DiffDrive.hh"
 
+#include <gz/msgs/boolean.pb.h>
 #include <gz/msgs/odometry.pb.h>
+#include <gz/msgs/pose_v.pb.h>
```
Comment (ahcorde) [inline src/systems/diff_drive/DiffDrive.cc:22]: pose too

File: src/systems/mecanum_drive/MecanumDrive.cc [hunk 12]
```diff
@@ -18,6 +18,7 @@
 #include "MecanumDrive.hh"
 
 #include <gz/msgs/odometry.pb.h>
+#include <gz/msgs/pose_v.pb.h>
```
Comment (ahcorde) [inline src/systems/mecanum_drive/MecanumDrive.cc:21]: Twist

File: src/systems/pose_publisher/PosePublisher.cc [hunk 13]
```diff
@@ -18,6 +18,7 @@
 #include "PosePublisher.hh"
 
 #include <gz/msgs/pose.pb.h>
+#include <gz/msgs/pose_v.pb.h>
```
Comment (ahcorde) [inline src/systems/pose_publisher/PosePublisher.cc:21]: Time

File: src/systems/scene_broadcaster/SceneBroadcaster.cc [hunk 14]
```diff
@@ -18,6 +18,8 @@
 #include "SceneBroadcaster.hh"
 
 #include <gz/msgs/scene.pb.h>
+#include <gz/msgs/pose_v.pb.h>
```
Comment (ahcorde) [inline src/systems/scene_broadcaster/SceneBroadcaster.cc:21]: StringMsg, SerializedStepMap, Model, Link, UInt32_V, Scene,Visual, Geometry, Material, Light, Sensor, CameraSensor, Distortion, LidarSensor, SensorNoise, IMUSensor, ParticleEmitter, Projector

File: src/systems/tracked_vehicle/TrackedVehicle.cc [hunk 15]
```diff
@@ -17,7 +17,9 @@
 
 #include "TrackedVehicle.hh"
 
+#include <gz/msgs/marker.pb.h>
```
Comment (ahcorde) [inline src/systems/tracked_vehicle/TrackedVehicle.cc:20]: double, twist, Vector3d, Time

File: src/systems/user_commands/UserCommands.cc [hunk 16]
```diff
@@ -30,6 +30,7 @@
 
 #include <gz/msgs/boolean.pb.h>
 #include <gz/msgs/entity_factory.pb.h>
+#include <gz/msgs/entity_factory_v.pb.h>
```
Comment (ahcorde) [inline src/systems/user_commands/UserCommands.cc:33]: entity, SphericalCoordinates, WheelSlipParametersCmd

File: test/integration/entity_system.cc [hunk 17]
```diff
@@ -16,6 +16,11 @@
  */
 
 #include <gtest/gtest.h>
+
+#include <gz/msgs/boolean.pb.h>
```
Comment (ahcorde) [inline test/integration/entity_system.cc:20]: Empty

File: test/integration/reset.cc [hunk 18]
```diff
@@ -17,6 +17,8 @@
 
 #include <gtest/gtest.h>
 
+#include <gz/msgs/world_control.pb.h>
```
Comment (ahcorde) [inline test/integration/reset.cc:20]: boolean

File: test/integration/scene_broadcaster_system.cc [hunk 19]
```diff
@@ -17,6 +17,10 @@
 
 #include <gtest/gtest.h>
 
+#include <gz/msgs/entity_factory.pb.h>
```
Comment (ahcorde) [inline test/integration/scene_broadcaster_system.cc:20]: Projector, ParticleEmitter, Scene, SerializedStepMap, StringMsg, Empty, boolean, UInt32_V

File: test/integration/sensors_system_update_rate.cc [hunk 20]
```diff
@@ -17,6 +17,8 @@
 
 #include <gtest/gtest.h>
 
+#include <gz/msgs/laserscan.pb.h>
```
Comment (ahcorde) [inline test/integration/sensors_system_update_rate.cc:20]: image

File: test/integration/spherical_coordinates.cc [hunk 21]
```diff
@@ -16,6 +16,9 @@
 */
 
 #include <gtest/gtest.h>
+
+#include <gz/msgs/entity_factory.pb.h>
```
Comment (ahcorde) [inline test/integration/spherical_coordinates.cc:20]: SphericalCoordinates, boolean

File: test/integration/tracked_vehicle_system.cc [hunk 22]
```diff
@@ -16,6 +16,11 @@
 */
 
 #include <gtest/gtest.h>
+
+#include <gz/msgs/odometry.pb.h>
```
Comment (ahcorde) [inline test/integration/tracked_vehicle_system.cc:20]: double

File: test/integration/triggered_publisher.cc [hunk 23]
```diff
@@ -18,6 +18,7 @@
 #include <gtest/gtest.h>
 
 #include <gz/msgs/empty.pb.h>
+#include <gz/msgs/int32_v.pb.h>
```
Comment (ahcorde) [inline test/integration/triggered_publisher.cc:21]: StringMsg, boolean, Vector2d, Int32, Float, Header

File: test/integration/user_commands.cc [hunk 24]
```diff
@@ -20,6 +20,7 @@
 #include <gz/msgs/entity_factory.pb.h>
```
Comment (ahcorde) [inline test/integration/user_commands.cc:20]: boolean, entity, pose, WheelSlipParametersCmd

File: test/integration/world_control_state.cc [hunk 25]
```diff
@@ -18,6 +18,8 @@
 #include <gtest/gtest.h>
 
 #include <gz/msgs/serialized_map.pb.h>
+#include <gz/msgs/world_control_state.pb.h>
```
Comment (ahcorde) [inline test/integration/world_control_state.cc:21]: Boolean

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: src/network/NetworkManagerPrimary.cc (hunk 7 above)
Comment (mjcarroll) [reply src/network/NetworkManagerPrimary.cc:31]: `SerializedStateMap` is in `serialized_map.pb.h`

File: src/network/NetworkManagerPrimary.hh (hunk 8 above)
Comment (mjcarroll) [reply src/network/NetworkManagerPrimary.hh:34]: `SerializedStateMap` is in `serialized_map.pb.h`

File: src/gui/plugins/component_inspector/ComponentInspector.cc [hunk 26]
```diff
@@ -15,6 +15,16 @@
...
 
+#include "ComponentInspector.hh"
+
+#include <gz/msgs/boolean.pb.h>
+#include <gz/msgs/entity_plugin_v.pb.h>
+#include <gz/msgs/light.pb.h>
+#include <gz/msgs/physics.pb.h>
+#include <gz/msgs/visual.pb.h>
+#include <gz/msgs/spherical_coordinates.pb.h>
+#include <gz/msgs/empty.pb.h>
```
Comment (iche033) [inline src/gui/plugins/component_inspector/ComponentInspector.cc:26]: alphabetize

File: test/integration/reset_sensors.cc [hunk 27]
```diff
@@ -17,7 +17,11 @@
 
 #include <gtest/gtest.h>
 
+#include <gz/msgs/boolean.pb.h>
+#include <gz/msgs/image.pb.h>
```
Comment (iche033) [inline test/integration/reset_sensors.cc:21]: alphabetize

File: src/network/NetworkManagerPrimary.cc [hunk 28]
```diff
@@ -17,6 +17,11 @@
 
 #include "NetworkManagerPrimary.hh"
 
+#include <gz/msgs/world_stats.pb.h>
+#include <gz/msgs/serialized_map.pb.h>
+#include "gz/sim/private_msgs/peer_control.pb.h"
+#include "gz/sim/private_msgs/simulation_step.pb.h"
```
Comment (iche033) [inline src/network/NetworkManagerPrimary.cc:23]: do you need `private_msgs/performer_affinity` here or is it ok that it's already included transitively by `simulation_step.pb.h`?

**Review by azeey — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]

**Comment by mjcarroll** [comment]
This isn't going to pass CI until we get one nightly turnover on msgs.  @azeey what do you want to do?

**Comment by azeey** [comment]
> This isn't going to pass CI until we get one nightly turnover on msgs. @azeey what do you want to do?

I manually triggered the build https://build.osrfoundation.org/job/gz-msgs10-debbuilder/446/, but the release package needs to be updated. See https://github.com/gazebo-release/gz-msgs10-release/pull/7

**Comment by azeey** [comment]
@osrf-jenkins run tests please

**Comment by mjcarroll** [comment]
```
 /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libgz-transport13.so.13.0.0~pre1: undefined reference to `gz::msgs::v10::ToString[abi:cxx11](gz::msgs::Discovery_Type const&)'
  /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libgz-fuel_tools9.so.9.0.0: undefined reference to `gz::msgs::v10::ConvertFuelMetadata(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, gz::msgs::FuelMetadata&)'
  /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libgz-fuel_tools9.so.9.0.0: undefined reference to `gz::msgs::v10::ConvertFuelMetadata(gz::msgs::FuelMetadata const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
  ```
  
  Since we moved symbols around, we probably need intermediate nightlies to run as well.  I will go ahead and trigger them now.

**Comment by azeey** [comment]
> ```
>  /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libgz-transport13.so.13.0.0~pre1: undefined reference to `gz::msgs::v10::ToString[abi:cxx11](gz::msgs::Discovery_Type const&)'
>   /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libgz-fuel_tools9.so.9.0.0: undefined reference to `gz::msgs::v10::ConvertFuelMetadata(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, gz::msgs::FuelMetadata&)'
>   /usr/bin/ld: /usr/lib/x86_64-linux-gnu/libgz-fuel_tools9.so.9.0.0: undefined reference to `gz::msgs::v10::ConvertFuelMetadata(gz::msgs::FuelMetadata const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
> ```
> 
> Since we moved symbols around, we probably need intermediate nightlies to run as well. I will go ahead and trigger them now.

Just finished doing that.


@osrf-jenkins run tests please

**Comment by azeey** [comment]
Windows is failing to build
```
 Failed to execute protoc compiler: Command '['C:/vcpkg/installed/x64-windows/tools/protobuf/protoc.exe', '--proto_path=C:/J/workspace/ign_gazebo-pr-win/ws/gz-sim/proto', '--descriptor_set_in=C:/J/workspace/ign_gazebo-pr-win/ws/install/gz-msgs10/share/gz/protos/gz-msgs10.gz_desc', '--plugin=protoc-gen-gzmsgs=C:/J/workspace/ign_gazebo-pr-win/ws/install/gz-msgs10//bin/gz-msgs10_protoc_plugin', '--cpp_out=dllexport_decl=GZ_MSGS_VISIBLE:C:/J/workspace/ign_gazebo-pr-win/ws/build/gz-sim8/gz-sim8-msgs_genmsg', '--gzmsgs_out=C:/J/workspace/ign_gazebo-pr-win/ws/build/gz-sim8/gz-sim8-msgs_genmsg', '--python_out=C:/J/workspace/ign_gazebo-pr-win/ws/build/gz-sim8/gz-sim8-msgs_genmsg/python/', 'C:/J/workspace/ign_gazebo-pr-win/ws/gz-sim/proto/gz/sim/private_msgs/peer_control.proto']' returned non-zero exit status 1.

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\CMakeFiles\db8da3a23e2d11963041e207edd372b4\peer_control.pb.h.rule;C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\CMakeFiles\db8da3a23e2d11963041e207edd372b4\peer_info.pb.h.rule;C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\CMakeFiles\db8da3a23e2d11963041e207edd372b4\performer_affinity.pb.h.rule;C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\CMakeFiles\db8da3a23e2d11963041e207edd372b4\simulation_step.pb.h.rule;C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\CMakeFiles\db8da3a23e2d11963041e207edd372b4\MessageTypes.hh.rule;C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\CMakeFiles\2ecabf52089c5108f2d458bd9c618598\gz-sim8-msgs.gz_desc.rule' exited with code -1. [C:\J\workspace\ign_gazebo-pr-win\ws\build\gz-sim8\proto\gz-sim8-msgs.vcxproj]
```

**Comment by azeey** [comment]
Rerunning Jenkins Jammy job to see if the failures are flaky: [Build Status](https://build.osrfoundation.org/job/gz_sim-ci-pr_any-jammy-amd64/161/)

**Comment by azeey** [comment]
@osrf-jenkins run tests please

**Comment by azeey** [comment]
[UserCommandsTest.PoseVector](https://build.osrfoundation.org/job/gz_sim-ci-pr_any-jammy-amd64/164/testReport/junit/(root)/UserCommandsTest/PoseVector/) is failing on both Linux and macOS. @mjcarroll can you take a look?

**Comment by mjcarroll** [comment]
> [UserCommandsTest.PoseVector](https://build.osrfoundation.org/job/gz_sim-ci-pr_any-jammy-amd64/164/testReport/junit/(root)/UserCommandsTest/PoseVector/) is failing on both Linux and macOS. @mjcarroll can you take a look?

Yep

**Comment by scpeters** [comment]
@osrf-jenkins run tests please

**Comment by azeey** [comment]
There's a build error on windows https://build.osrfoundation.org/job/ign_gazebo-pr-win/6095/

**Comment by scpeters** [comment]
https://build.osrfoundation.org/job/gz_sim-ci-pr_any-jammy-amd64/ is not being started on the most recent commits

**Comment by azeey** [comment]
Merging this and https://github.com/gazebosim/gz-sim/pull/2100 even though there's a build error on windows to unblock other PRs. We'll hopefully have the windows fix later today.

**Pull request merged**